.github/workflows/k8s-deploy.yml

```
@@ -0,0 +1,350 @@
+    - name: Deploy to staging
+      run: |
+        kustomize build k8s/environments/staging | kubectl apply -f -
+        kubectl rollout status deployment/walmart-grocery-agent -n crewai-team-staging --timeout=600s
+
+    - name: Run smoke tests
+      run: |
+        kubectl wait --for=condition=ready pod -l app=walmart-grocery-agent -n crewai-team-staging --timeout=300s
+        
+        # Get service endpoint
+        SERVICE_IP=$(kubectl get service walmart-grocery-service -n crewai-team-staging -o jsonpath='{.status.loadBalancer.ingress[0].ip}')
+        
```

YAMLlint (1.37.1)

[error] trailing spaces

(trailing-spaces)